_episodes/01-introduction.md

```diff
@@ -1,15 +1,183 @@
+length = len(energy_kcal)
+print('The length of this list is', length)
+```
+{: .language-python}
+
+```
+The length of this list is 4
+```
+{: .output}
+
+If I want to operate on a particular element of the list, you use the list name and then put in brackets which element of the list you want.  **In python counting starts at zero.  So the first element of the list is `list[0]`**
+
+```
+print(energy_kcal[1])
+```
+{: .language-python}
+
+```
+-2.7
+```
+{: .output}
 
+```
+energy_kcal[1] = -2.9
+print(energy_kcal)
+```
+{: .language-python}
+
+```
+[-13.4, -2.9, 5.4, 42.1]
+```
+{: .output}
+
+## For loops
+Often, you will want to do something to every element of a list.  The structure
+to do this is called a `for` loop.  The general structure of a `for` loop is
+```
+for variable in list:
+    do things using variable
+```
+{: .language-python}
+
+Indentation is very important in python.  There is nothing like an `end` or `exit` statement that tells you that you are finished with the loop.  The indentation shows you what statements are in the loop.  Let's use a loop to change all of our energies in kcal to kJ.
+```
+for number in energy_kcal:
+    kJ = number*4.184
+    print(kJ)
+```
+{: .language-python}
+
+```
+-56.0656
+-12.1336
+22.593600000000002
+176.1464
+```
+{: .output}
+
+Now it seems like we are really getting somewhere with our program!  But it would be even better if instead of just printing the values, it saved them in a new list.  To do this, we are going to use the `append` function.  The `append` function adds a new item to the end of an existing list.  The general form of the append function is
+```
+list_name.append(new_thing)
+```
+{: .language-python}
+
+Try running this block of code.  See if you can figure out why it doesn't work.
+```
+for number in energy_kcal:
+    kJ = number*4.184
+    energy_kJ.append(kJ)
+
+print(energy_kJ)
+```
+{: .language-python}
+
+This code doesn't work because on the first iteration of our loop, the list `energy_kJ` doesn't exist.  To make it work, we have to start the list outside of the loop.  The list can be blank when we start it, but we have to start it.
+
+```
+energy_kJ = []
+for number in energy_kcal:
+    kJ = number*4.184
+    energy_kJ.append(kJ)
+
+print(energy_kJ)
+```
+{: .language-python}
+
+```
+[-56.0656, -12.1336, 22.593600000000002, 176.1464]
+```
+{: .output}
+
+## A note about jupyter notebooks
+If you use the jupyter notebook for your python interpreter, the notebook only executes the current code block.  If you change a value and then go back and run an earlier code block, it will use the new value, which may give you incorrect analysis.  Similarly, if you open your jupyter notebook later, and try to run a code block in the middle, it may tell you that your variables are undefined, even though you can clearly see them defined in earlier code blocks.  But if you didn't re-run those code blocks, then python doesn't know they exist.  
+
+Add some exercises here.  
```

Comment:
We could consider having another lesson with an intro to Jupyter notebook. I think SW carpentry has one of these we could borrow heavily from. Or, it could be included as part of set up.